src/assets/JSAsset.js

```diff
@@ -77,19 +78,38 @@ class JSAsset extends Asset {
 
     // Check if there is a babel config file. If so, determine which parser plugins to enable
     this.babelConfig = await babel.getConfig(this);
-    Object.assign(options, this.babelConfig);
-    // if (this.babelConfig) {
-    //   const file = new BabelFile(this.babelConfig);
-    //   options.plugins.push(...file.parserOpts.plugins);
-    // }
+    if (this.babelConfig) {
+      if (this.babelConfig.babelVersion === 6) {
+        options.parserOpts.tokens = true;
+        options = options.parserOpts;
+      }
+
+      Object.assign(options, this.babelConfig);
+
+      let babelVersion = this.babelConfig.babelVersion;
+      Object.defineProperty(options, 'babelVersion', {
+        value: babelVersion,
+        configurable: true
+      });
+
+      // if (this.babelConfig) {
+      //   const file = new BabelFile(this.babelConfig);
+      //   options.plugins.push(...file.parserOpts.plugins);
+      // }
+    }
 
     return options;
   }
 
   async parse(code) {
     const options = await this.getParserOptions();
-    // return babylon.parse(code, options);
-    return babelCore.parse(code, options);
+
+    if (options.babelVersion === 6) {
+      let babylon = await localRequire('babylon', this.name);
```

Comment:
Why do we still need babylon? I think we should always parse with babel 7 and convert the AST as needed. Our internal plugins will all be using babel 7, and some of them run before custom babel transforms.

Reply:
I left it there for plugins that run on ast creation. Not sure if that's really necessary